packages/store/src/internal/state-factory.ts

```diff
@@ -218,6 +215,22 @@ export class StateFactory {
     return forkJoin(results);
   }
 
+  private getUnmountedStateList(stateClasses: StateClass[]): StateClass[] {
+    const unmountedStateList: StateClass[] = [];
+    const statesMap: StatesByName = this.statesByName;
+
+    for (const stateClass of stateClasses) {
+      const stateName: string = StoreValidators.checkStateNameIsUnique(stateClass, statesMap);
+      const unmountedState: boolean = !statesMap[stateName];
+      if (unmountedState) {
+        unmountedStateList.push(stateClass);
+        statesMap[stateName] = stateClass;
```

Comment:
This method is confusing because it has side effects of modifying the `this.statesByName` but it is named `get...` which implies determinism but does not fulfill that promise.

Reply:
Fixed now by naming the method `addToStatesMap` and adding a descriptive return value.

Reply:
Goog catch